### PR TITLE
Release notes for 2.6.0

### PR DESCRIPTION
**Pull request by abrookins**
[Rendered](https://github.com/PrefectHQ/prefect/blob/release-2.6.0/RELEASE-NOTES.md#release-260)

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,120 @@
 # Prefect Release Notes
 
+## Release 2.6.0
+
+### First-class configuration of results 🎉
+
+Previously, Prefect serialized the results of all flows and tasks with pickle then wrote them to your local file system.
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:7]:
```suggestion
Previously, Prefect serialized the results of all flows and tasks with pickle, then wrote them to your local file system.
```

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,120 @@
...
+In this release, we're excited to announce this behavior is fully configurable and customizable.
+
+Here are some highlights:
+
+- Persistence of results is off by default.
+    - We will turn on result persistence automatically if needed for a feature you're using but you can always opt-out.
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:13]:
```suggestion
    - We will turn on result persistence automatically if needed for a feature you're using, but you can always opt-out.
```

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,120 @@
...
+    - You can easily opt-in for any flow or task.
+- You can choose the result serializer.
+    - By default, we continue to use a pickle serializer, now with the ability to choose a custom implementation.
+    - We now offer a JSON result serializer with support for all of the types supported by Pydantic.
+    - You can also write your own serializer for full control.
+    - Unless your results are being persisted, they will not be serialized.
+- You can change the result storage.
+    - By default, we will continue to use the local file system.
+    - You can specify any of our storage blocks i.e. AWS S3.
```
Comment (tpdorsey) [inline RELEASE-NOTES.md:22]:
```suggestion
    - You can specify any of our storage blocks, such as AWS S3.
```

**Review by tpdorsey — APPROVED** [review]
Mostly commas. Looks good

**Review by billpalombi — APPROVED** [review]
LGTM

**Pull request merged**